ibr/ibr-sdk/src/index.js

```diff
@@ -183,36 +180,44 @@ function drawSingleStructureSidebar(structure, structureName) {
 }
 
 /**
- * Parse top level of decoded IBR Object into structures.
- * @param {binary} ibrRawData Raw data from IBR binary data file.
- * @param {number} structureIndex Index of current structure(floor).
- * @param {HTMLElement} parentElement parent HTML element that the
-  visualization will be append on.
- */
-function render(ibrRawData, structureIndex, parentElement) {
-  scene = generateScene(parentElement);
+   * Initialize IBRObject from input IBR raw data in binary format.
+   * @param {Buffer} ibrRawData binary IBR data from uploaded IBR file.
+   * @return {IBRObject} ibrObject IBRObject generated from input binary.
+   */
+function init(ibrRawData) {
   const ibrData = InternalBuildingRepresentation.read(
       new Pbf(ibrRawData));
   const ibrObject = new IBRObject( ibrData );
-  const structure = {};
-  // Visualization layers of current structure
-  structure['layers'] = renderLayer( ibrObject,
-      structureIndex, scene );
-  // Sub-structures of the current structure
-  structure['structures'] = ibrObject.getSubStructures();
+  return ibrObject;
+}
+
+/**
+   * Enable Export and parse top level of decoded IBR Object into structures.
+   * @param {binary} ibrObject IBRObject created from input IBR binary data
+   file.
+   * @param {number} structureIndex Index of current structure(floor).
+   * @param {HTMLElement} parentElement parent HTML element that the
+    visualization will be append on.
+   */
+function render(ibrObject, structureIndex, parentElement) {
+  scene = generateScene(parentElement);
+  document.getElementById('dwn-btn').style.display = 'block';
+  document.getElementById('filename').style.display = 'block';
+  // Visualization visualizations of current structure
+  renderVisualization( ibrObject, structureIndex, scene );
 }
 
 /**
  * Parse current IBR Object into structures.
  * @param {IBRObject} ibrObject IBRObject created from current structure data.
  * @param {number} structureIndex Index of current structure(floor).
- * @return {Map.<String, List.<Object>>} Map of list of layers and
+ * @return {Map.<String, List.<Object>>} Map of list of visualizations and
  structures.
  */
 function renderSingleIBRStructure(ibrObject, structureIndex) {
   const structure = {};
-  // Visualization layers of current structure
-  structure['layers'] = renderLayer( ibrObject,
+  // Visualization visualizations of current structure
```

Comment:
nit: in the future, you could probably put all the s/layer/visualization/g changes in a different separate PR, it makes it easier to compare diffs

Reply:
Ok!